### PR TITLE
8294772: Remove workaround in os::dll_address_to_library_name

### PR DESCRIPTION
**Pull request by jdksjolen**
I ported the BSD implementation of this function. According to the comment this bug should be fixed, as the oldest glibc I can see us supporting is 2.12.x. I've made an attempt at finding the bug that the comment refers to on glibc's bug tracker but has been unsuccesful in this.

The second commit deletes what seems like an outdated comment (in os_bsd this refers to a code snippet right below the comment).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294772](https://bugs.openjdk.org/browse/JDK-8294772): Remove workaround in os::dll_address_to_library_name

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [7ae05ca1](https://git.openjdk.org/jdk/pull/10559/files/7ae05ca176e1b9b0cd0defd5a24c17e0bd2d92b5)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to [7ae05ca1](https://git.openjdk.org/jdk/pull/10559/files/7ae05ca176e1b9b0cd0defd5a24c17e0bd2d92b5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10559/head:pull/10559` \
`$ git checkout pull/10559`

Update a local copy of the PR: \
`$ git checkout pull/10559` \
`$ git pull https://git.openjdk.org/jdk pull/10559/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10559`

View PR using the GUI difftool: \
`$ git pr show -t 10559`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10559.diff">https://git.openjdk.org/jdk/pull/10559.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
This looks good to me. Thanks for cleaning this up.

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -1372,84 +1372,22 @@ bool os::dll_address_to_function_name(address addr, char *buf,
...
-    d->base = libbase;
-    if (d->fname) {
-      jio_snprintf(d->fname, d->buflen, "%s", info->dlpi_name);
-    }
-    return 1;
-  }
-  return 0;
-}
-
+// Ported from BSD version
```
Comment (iklam) [inline src/hotspot/os/linux/os_linux.cpp:1375]: I think this comment is not necessary. In fact, the corresponding comment `// ported from solaris version` in os_bsd.cpp should probably be removed as well.

**Review by iklam — APPROVED** [review]
Looks good except for a nit on the comment.

File: src/hotspot/os/linux/os_linux.cpp (hunk 1 above)
Comment (jdksjolen) [reply src/hotspot/os/linux/os_linux.cpp]: Sounds good, removed them.

**Comment by jdksjolen** [comment]
/integrate

@dholmes-ora , @iklam. I suspect that I will need new approvals from you to be allowed to integrate.

**Comment by iklam** [comment]
What kind of testing have you done? I saw that the GitHub Actions included only jcheck.

**Comment by iklam** [comment]
> What kind of testing have you done? I saw that the GitHub Actions included only jcheck.

@jdksjolen has run tiers 1-3 and builds-tier-5 in our CI. This PR is ready to be integrated.

/sponsor

**Pull request closed without merging**